File: src/nodejs/src/dtos/ApiCode.ts
/**
 * Enum representing standard API response codes.
 */
export enum ApiCode {
  /** Operation completed successfully. */
  Success = 1,
  /** An unknown error occurred. */
  UnknownError = 2,
  /** The request was invalid or malformed. */
  InvalidRequest = 3,
  /** No data was found for the requested resource. */
  NoDataFound = 4,
  /** An error occurred while accessing the data store. */
  DataAccessError = 5,
  /** The operation failed to complete. */
  OperationFailed = 6,
  /** The operation timed out. */
  OperationTimeout = 7,
}

File: src/python/app/schemas/responses/api_code.py
"""API response codes enum."""

from enum import IntEnum


class ApiCode(IntEnum):
    """API response codes matching C# enum."""

    SUCCESS = 1
    UNKNOWN_ERROR = 2
    INVALID_REQUEST = 3
    NO_DATA_FOUND = 4
    DATA_ACCESS_ERROR = 5
    OPERATION_FAILED = 6
    OPERATION_TIMEOUT = 7
